src/h5web/nexus-visualizer/NexusVisualizer.tsx

```diff
@@ -1,17 +1,70 @@
 import React, { ReactElement } from 'react';
 import type { HDF5Group } from '../providers/models';
+import { getSignalLink, getInterpretationVis } from './utils';
 import styles from '../Visualizer.module.css';
+import { useEntity } from '../providers/hooks';
+import { isDataset } from '../providers/utils';
+import VisSelector from '../dataset-visualizer/VisSelector';
+import VisDisplay from '../dataset-visualizer/VisDisplay';
+import { VIS_DEFS } from '../visualizations';
+import { getSupportedVis } from '../dataset-visualizer/utils';
 
 interface Props {
   group: HDF5Group;
 }
 
 function NexusVisualizer(props: Props): ReactElement {
   const { group } = props;
+
+  const signalLink = getSignalLink(group);
+  const signalDataset = useEntity(signalLink);
+
+  if (!signalLink) {
+    return (
+      <div className={styles.visualizer}>
+        <div className={styles.visBar} />
+        <p className={styles.fallback}>
+          NXdata group has no attribute signal or it points to an non-existing
+          dataset.
```

Comment:
We might want to break down `getSignalLink` to be able to give the real cause of the "error". See my other comment on `signalLink`.

Reply:
Ah yes, in this case you may want to inline the code of `getSignalLink`.

I don't hink you need all those `div` wrappers and the vis bar. It should be fine to just return the fallback text.

Reply:
perhaps a little `getAttributeValue()` would help keep things clean